blocks/testing/include/gnuradio-4.0/testing/FunctionBlocks.hpp

```diff
@@ -0,0 +1,108 @@
+#ifndef GNURADIO_TESTING_FUNCTION_BLOCKS_HPP
+#define GNURADIO_TESTING_FUNCTION_BLOCKS_HPP
+
+#include <fmt/format.h>
+#include <fmt/ranges.h>
+
+#include <gnuradio-4.0/Block.hpp>
+#include <gnuradio-4.0/reflection.hpp>
+#include <gnuradio-4.0/Tag.hpp>
+
+namespace gr::testing {
+
+template<typename T>
+struct FunctionSource : public gr::Block<FunctionSource<T>> {
```

Comment:
These classes need some documentation w.r.t. what they do and what they are supposed to be used for.